src/Language/Fixpoint/Types/Names.hs

```diff
@@ -518,7 +518,7 @@ tidySymbol s
   | otherwise = s''
   where
     s'        = unPrefixSymbol kArgPrefix s
-    s''       = unPrefixSymbol symSepName . unPrefixSymbol hvarPrefix $ s'
+    s''       = consSym '$' . unPrefixSymbol symSepName . unSuffixSymbol . unPrefixSymbol hvarPrefix $ s'
```

Comment:
I'm out of my depth here, but I just wanted to suggest adding a comment to `tidySymbol` explaining what it is meant to do.

Reply:
And similar suggestion for the other functions modified in the PR. Alternatively, perhaps the source code could be made more self explanatory if the structure of the symbols is explained in a comment.